### PR TITLE
add support for CALL statements

### PR DESCRIPTION
**Pull request by lovasoa**
see https://dev.mysql.com/doc/refman/8.0/en/call.html

Adds support for MySQL's CALL statements 
(stored procedure invocation)

This one is very important for [SQLPage](https://github.com/lovasoa/SQLpage) too,
because it allows working around other unsupported syntax
by wrapping logic in stored procedures.

### CODE REVIEW

File: src/ast/mod.rs [hunk 1]
```diff
@@ -1715,7 +1716,9 @@ pub enum Statement {
     ///
     /// Note: this is a PostgreSQL-specific statement,
     /// but may also compatible with other SQL.
-    Discard { object_type: DiscardObject },
+    Discard {
+        object_type: DiscardObject,
+    },
```
Comment (lovasoa) [inline src/ast/mod.rs:1721]: @alamb , the reformatting is made by cargo fmt

File: tests/sqlparser_mysql.rs [hunk 2]
```diff
@@ -1817,6 +1817,14 @@ fn parse_hex_string_introducer() {
...
 }
 
+#[test]
+fn parse_call() {
+    mysql().verified_stmt("CALL my_procedure()");
+    mysql().verified_stmt("CALL my_procedure(1, 'a')");
+    mysql().verified_stmt("CALL my_procedure(1, 'a', @my_var)");
+    mysql().verified_stmt("CALL my_procedure");
+}
+
```
Comment (universalmind303) [inline tests/sqlparser_mysql.rs:1827]: It doesn't look like the `CALL` keyword is gated to `mysql` dialect at all. Can we either add some tests to generic, or gate it behind the mysql dialect? 

I'd prefer adding it to generic as other dialects such as duckdb also support `CALL`.

Comment (lovasoa) [reply tests/sqlparser_mysql.rs]: You are right ! I changed the tests.

**Review by alamb — APPROVED** [review]
Looks good to me -- thank you @lovasoa

**Pull request merged**